src/main/java/com/amazon/jenkins/ec2fleet/CloudNanny.java

```diff
@@ -77,4 +93,12 @@ protected void doRun() {
     private static List<Cloud> getClouds() {
         return Jenkins.getActiveInstance().clouds;
     }
+
+    @VisibleForTesting
+    private AtomicInteger getRecurrenceCounter(EC2FleetCloud fleetCloud) {
+        if (!recurrenceCounters.containsKey(fleetCloud)) {
+            recurrenceCounters.putIfAbsent(fleetCloud, new AtomicInteger(fleetCloud.getCloudStatusIntervalSec()));
+        }
+        return recurrenceCounters.get(fleetCloud);
```

Comment:
It's not safe to check contains, abd rgen call ```get``` as key could be deleted in between. I siuggest to take value and check if not null.

Reply:
The other way to implement this:

```java
AtomicInteger counter = new AtomicInteger(fleetCloud.getCloudStatusIntervalSec());
AtomicInteger result = referenceCounters.putIfAbsent(fleetCloud, counter);
return result == null ? counter : result;
```

requires a creation of a new atomic integer every time. Admittedly this is only once per second and it never makes it out of eden space if unused. Is that the way you would prefer this?

Reply:
(in Java 8, there is `computeIfAbsent`, however this plugin is still using java 7 language compat).